components/logins/ffi/src/lib.rs

```diff
@@ -85,39 +87,41 @@ unsafe fn bytes_to_key_string(key_bytes: *const u8, len: usize) -> Option<String
 /// # Safety
 ///
 /// Dereferences the `encryption_key` pointer, and is thus unsafe.
-#[no_mangle]
-pub unsafe extern "C" fn sync15_passwords_state_new_with_hex_key(
-    db_path: FfiStr<'_>,
-    encryption_key: *const u8,
-    encryption_key_len: u32,
-    error: &mut ExternError,
-) -> u64 {
-    log::debug!("sync15_passwords_state_new_with_hex_key");
-    STORES.insert_with_result(error, || -> logins::Result<_> {
-        let path = db_path.as_str();
-        let key = bytes_to_key_string(encryption_key, encryption_key_len as usize);
-        // We have a Option<String>, but need an Option<&str>...
-        let opt_key_ref = key.as_deref();
-        Ok(Arc::new(Mutex::new(PasswordStore::new(path, opt_key_ref)?)))
-    })
-}
+// TODO_WILL_NOT_IMPLEMENT
+// #[no_mangle]
+// pub unsafe extern "C" fn sync15_passwords_state_new_with_hex_key(
+//     db_path: FfiStr<'_>,
+//     encryption_key: *const u8,
+//     encryption_key_len: u32,
+//     error: &mut ExternError,
+// ) -> u64 {
+//     log::debug!("sync15_passwords_state_new_with_hex_key");
+//     STORES.insert_with_result(error, || -> logins::Result<_> {
+//         let path = db_path.as_str();
+//         let key = bytes_to_key_string(encryption_key, encryption_key_len as usize);
+//         // We have a Option<String>, but need an Option<&str>...
+//         let opt_key_ref = key.as_deref();
+//         Ok(Arc::new(Mutex::new(PasswordStore::new(path, opt_key_ref)?)))
+//     })
+// }
 
 /// Opens an existing database that stores its salt in the header bytes and retrieves its salt.
 ///
 /// # Safety
 ///
 /// Dereferences the `encryption_key` pointer, and is thus unsafe.
-#[no_mangle]
-pub unsafe extern "C" fn sync15_passwords_get_db_salt(
-    db_path: FfiStr<'_>,
-    encryption_key: FfiStr<'_>,
-    error: &mut ExternError,
-) -> *mut c_char {
-    log::debug!("sync15_passwords_get_db_salt");
-    let path = db_path.as_str();
-    let key = encryption_key.as_str();
-    ffi_support::call_with_result(error, || LoginDb::open_and_get_salt(path, key))
-}
+// #[no_mangle]
```

Comment:
I think it would be helpful to understand the state if you actually removed entirely functions which you believe have been "fully" converted in a sane way - which seems to be most of them. Then what remains in this file can be the "things we need to think about a bit more".

Alternatively, moving some comments to the UDL might make more sense, as that is where the questions will need to be answered, and will allow you to completely delete the `logins/ffi` directory, which will be quite a milestone :)